crates/muvm/src/guest/bin/muvm-guest.rs

```diff
@@ -81,12 +80,7 @@ fn main() -> Result<()> {
     let pulse_path = pulse_path.join("native");
     setup_socket_proxy(pulse_path, 3333)?;
 
-    #[cfg(feature = "x11bridge")]
-    let direct_x11 = setup_x11_forwarding(run_path)?;
-    #[cfg(not(feature = "x11bridge"))]
-    let direct_x11 = false;
-
-    if !direct_x11 {
+    if !setup_x11_forwarding(run_path)? {
```

Comment:
I think this function shouldn't return bool, but just propagate the error if something is wrong?

Reply:
It returns `Result<bool>`. `Err(x)` means something is wrong, `Ok(true)` means X11 forwarding is enabled, `Ok(false)` means X11 forwarding is disabled (and sommelier should be used instead).

Reply:
Yes, I understand. But I think it should rather just error instead of falling back to sommelier automatically? The user can be expected to try with `--sommelier` flag if the default doesn't work...

Reply:
Hmm... I need to look at the code again. Do we have an actual way to tell whether x11bridge is explicitly disabled (i.e. passing the `--sommelier` flag), vs. the case of it just failing up be set up? I think the latter should result in an error instead.

Reply:
Uhh... Never mind. `HOST_DISPLAY` is only set when `--sommelier` flag is not being passed.